// temboo/Library/Google/Calendar.js

var util = require("util");
var choreography = require("temboo/core/choreography");


/*
    CreateCalendar
    Create a new secondary calendar.
*/


var CreateCalendar = function(session) {
    /*
        Create a new instance of the CreateCalendar Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/CreateCalendar"
    CreateCalendar.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new CreateCalendarResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new CreateCalendarInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the CreateCalendar
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var CreateCalendarInputSet = function() {
    CreateCalendarInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the Description input for this Choreo. ((optional, string) A description of the calendar.)
        */
        this.set_Description = function(value) {
            this.setInput("Description", value);
        }

        /*
        Set the value of the Location input for this Choreo. ((optional, string) Geographic location of the calendar such as "Los Angeles" or "New York".)
        */
        this.set_Location = function(value) {
            this.setInput("Location", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the Timezone input for this Choreo. ((optional, string) The timezone for the new calendar, such as "America/Los_Angeles". Defaults to UTC if left blank.)
        */
        this.set_Timezone = function(value) {
            this.setInput("Timezone", value);
        }

        /*
        Set the value of the Title input for this Choreo. ((required, string) The name for the new calendar.)
        */
        this.set_Title = function(value) {
            this.setInput("Title", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the CreateCalendar Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var CreateCalendarResultSet = function(resultStream) {
    CreateCalendarResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Represents a Google Calendar Resource.
    */
    this.getCalendar = function()  {
    	return new  GoogleCalendar(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(CreateCalendar, choreography.Choreography);
util.inherits(CreateCalendarInputSet, choreography.InputSet);
util.inherits(CreateCalendarResultSet, choreography.ResultSet);
exports.CreateCalendar = CreateCalendar;


/*
    CreateEvent
    Create a new event in a specified calendar.
*/


var CreateEvent = function(session) {
    /*
        Create a new instance of the CreateEvent Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/CreateEvent"
    CreateEvent.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new CreateEventResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new CreateEventInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the CreateEvent
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var CreateEventInputSet = function() {
    CreateEventInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The ID for the calendar in which to add the event.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the EndDate input for this Choreo. ((required, string) The end date of the event, in the format "2012-04-10".)
        */
        this.set_EndDate = function(value) {
            this.setInput("EndDate", value);
        }

        /*
        Set the value of the EndTime input for this Choreo. ((required, string) The end time for the event, in the format "10:30:00".)
        */
        this.set_EndTime = function(value) {
            this.setInput("EndTime", value);
        }

        /*
        Set the value of the EventDescription input for this Choreo. ((optional, string) A short description of the event.)
        */
        this.set_EventDescription = function(value) {
            this.setInput("EventDescription", value);
        }

        /*
        Set the value of the EventLocation input for this Choreo. ((optional, string) The location for the new event.)
        */
        this.set_EventLocation = function(value) {
            this.setInput("EventLocation", value);
        }

        /*
        Set the value of the EventTitle input for this Choreo. ((required, string) The title for the new event.)
        */
        this.set_EventTitle = function(value) {
            this.setInput("EventTitle", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the StartDate input for this Choreo. ((required, string) The start date of the event, in the format "2012-11-03".)
        */
        this.set_StartDate = function(value) {
            this.setInput("StartDate", value);
        }

        /*
        Set the value of the StartTime input for this Choreo. ((required, string) The start time for the event, in the format "10:00:00".)
        */
        this.set_StartTime = function(value) {
            this.setInput("StartTime", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the CreateEvent Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var CreateEventResultSet = function(resultStream) {
    CreateEventResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "CreateEvent" output from this Choreo execution. (The request template with appropriate inputs passed.)
        */
        this.get_CreateEvent = function() {
            return this.getResult("CreateEvent");
        }
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
        /*
        Retrieve the value for the "TimezoneSetting" output from this Choreo execution. ((string) The timezone setting retrieved from the specified calendar.)
        */
        this.get_TimezoneSetting = function() {
            return this.getResult("TimezoneSetting");
        }
    /*
    A Google Calendar Event resource.
    */
    this.getEvent = function()  {
    	return new  GoogleEvent(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(CreateEvent, choreography.Choreography);
util.inherits(CreateEventInputSet, choreography.InputSet);
util.inherits(CreateEventResultSet, choreography.ResultSet);
exports.CreateEvent = CreateEvent;


/*
    DeleteAllEvents
    Delete all events from a primary calendar of an account. Note that this operation can't be undone.
*/


var DeleteAllEvents = function(session) {
    /*
        Create a new instance of the DeleteAllEvents Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/DeleteAllEvents"
    DeleteAllEvents.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new DeleteAllEventsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new DeleteAllEventsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the DeleteAllEvents
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var DeleteAllEventsInputSet = function() {
    DeleteAllEventsInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the DeleteAllEvents Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var DeleteAllEventsResultSet = function(resultStream) {
    DeleteAllEventsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (No content is returned for delete calendar operations.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
}

util.inherits(DeleteAllEvents, choreography.Choreography);
util.inherits(DeleteAllEventsInputSet, choreography.InputSet);
util.inherits(DeleteAllEventsResultSet, choreography.ResultSet);
exports.DeleteAllEvents = DeleteAllEvents;


/*
    DeleteCalendar
    Delete a secondary calendar.
*/


var DeleteCalendar = function(session) {
    /*
        Create a new instance of the DeleteCalendar Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/DeleteCalendar"
    DeleteCalendar.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new DeleteCalendarResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new DeleteCalendarInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the DeleteCalendar
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var DeleteCalendarInputSet = function() {
    DeleteCalendarInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The ID for the calendar to delete.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((required, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((required, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((required, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the DeleteCalendar Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var DeleteCalendarResultSet = function(resultStream) {
    DeleteCalendarResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (No content is returned for delete calendar operations.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
}

util.inherits(DeleteCalendar, choreography.Choreography);
util.inherits(DeleteCalendarInputSet, choreography.InputSet);
util.inherits(DeleteCalendarResultSet, choreography.ResultSet);
exports.DeleteCalendar = DeleteCalendar;


/*
    DeleteEvent
    Delete a specific event from a specified calendar.
*/


var DeleteEvent = function(session) {
    /*
        Create a new instance of the DeleteEvent Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/DeleteEvent"
    DeleteEvent.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new DeleteEventResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new DeleteEventInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the DeleteEvent
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var DeleteEventInputSet = function() {
    DeleteEventInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The ID for the calendar to delete.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The name of the calendar that you want to retrieve information for. Note that if there are multiple calendars with the same name, only the first one will be returned.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the EventID input for this Choreo. ((required, string) The unique ID for the event to delete.)
        */
        this.set_EventID = function(value) {
            this.setInput("EventID", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the DeleteEvent Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var DeleteEventResultSet = function(resultStream) {
    DeleteEventResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "AccessToken" output from this Choreo execution. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.get_AccessToken = function() {
            return this.getResult("AccessToken");
        }
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (No content is returned for delete calendar operations.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
}

util.inherits(DeleteEvent, choreography.Choreography);
util.inherits(DeleteEventInputSet, choreography.InputSet);
util.inherits(DeleteEventResultSet, choreography.ResultSet);
exports.DeleteEvent = DeleteEvent;


/*
    GetAllCalendars
    Retrieve data for all calendars.
*/


var GetAllCalendars = function(session) {
    /*
        Create a new instance of the GetAllCalendars Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/GetAllCalendars"
    GetAllCalendars.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetAllCalendarsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetAllCalendarsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetAllCalendars
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetAllCalendarsInputSet = function() {
    GetAllCalendarsInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the Count input for this Choreo. ((optional, integer) The maximum number of calendars to return. The default is 15.)
        */
        this.set_Count = function(value) {
            this.setInput("Count", value);
        }

        /*
        Set the value of the Fields input for this Choreo. ((optional, string) The calendar fields to return. Accepted values are "etag", "items", "kind", and "nextPageToken". All calendar items are returned by default.)
        */
        this.set_Fields = function(value) {
            this.setInput("Fields", value);
        }

        /*
        Set the value of the Hidden input for this Choreo. ((optional, boolean) Enter "1" to return hidden calendars in the results, or "0" (the default) to not return them.)
        */
        this.set_Hidden = function(value) {
            this.setInput("Hidden", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetAllCalendars Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetAllCalendarsResultSet = function(resultStream) {
    GetAllCalendarsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    A list of Google Calendars.
    */
    this.getCalendars = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["items"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  GoogleCalendarListEntry(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(GetAllCalendars, choreography.Choreography);
util.inherits(GetAllCalendarsInputSet, choreography.InputSet);
util.inherits(GetAllCalendarsResultSet, choreography.ResultSet);
exports.GetAllCalendars = GetAllCalendars;


/*
    GetAllEvents
    Retrieve data for all events in a specified calendar.
*/


var GetAllEvents = function(session) {
    /*
        Create a new instance of the GetAllEvents Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/GetAllEvents"
    GetAllEvents.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetAllEventsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetAllEventsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetAllEvents
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetAllEventsInputSet = function() {
    GetAllEventsInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The unique ID for the calendar with the events to retrieve.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetAllEvents Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetAllEventsResultSet = function(resultStream) {
    GetAllEventsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get All Google Calendar Events
    */
    this.getEventList = function()  {
    	return new  GoogleEventList(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(GetAllEvents, choreography.Choreography);
util.inherits(GetAllEventsInputSet, choreography.InputSet);
util.inherits(GetAllEventsResultSet, choreography.ResultSet);
exports.GetAllEvents = GetAllEvents;


/*
    GetAllSettings
    Retrieves all settings for a Google Calendar account.
*/


var GetAllSettings = function(session) {
    /*
        Create a new instance of the GetAllSettings Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/GetAllSettings"
    GetAllSettings.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetAllSettingsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetAllSettingsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetAllSettings
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetAllSettingsInputSet = function() {
    GetAllSettingsInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the Fields input for this Choreo. ((optional, string) The settings fields to return. Returns etag, item IDs, and item values by default.)
        */
        this.set_Fields = function(value) {
            this.setInput("Fields", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetAllSettings Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetAllSettingsResultSet = function(resultStream) {
    GetAllSettingsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get All Google Calendar Settings
    */
    this.getSettingList = function()  {
    	return new  GoogleSettingList(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(GetAllSettings, choreography.Choreography);
util.inherits(GetAllSettingsInputSet, choreography.InputSet);
util.inherits(GetAllSettingsResultSet, choreography.ResultSet);
exports.GetAllSettings = GetAllSettings;


/*
    GetCalendar
    Retrieve data for a specific calendar in a Google account.
*/


var GetCalendar = function(session) {
    /*
        Create a new instance of the GetCalendar Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/GetCalendar"
    GetCalendar.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetCalendarResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetCalendarInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetCalendar
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetCalendarInputSet = function() {
    GetCalendarInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The ID for the calendar to retrieve.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetCalendar Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetCalendarResultSet = function(resultStream) {
    GetCalendarResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Represents a Google Calendar Resource.
    */
    this.getCalendar = function()  {
    	return new  GoogleCalendar(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(GetCalendar, choreography.Choreography);
util.inherits(GetCalendarInputSet, choreography.InputSet);
util.inherits(GetCalendarResultSet, choreography.ResultSet);
exports.GetCalendar = GetCalendar;


/*
    GetEvent
    Retrieve data for all calendars in an account.
*/


var GetEvent = function(session) {
    /*
        Create a new instance of the GetEvent Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/GetEvent"
    GetEvent.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetEventResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetEventInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetEvent
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetEventInputSet = function() {
    GetEventInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The unique ID for the calendar with the events to retrieve.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the EventID input for this Choreo. ((required, string) The unique ID for the event to retrieve.)
        */
        this.set_EventID = function(value) {
            this.setInput("EventID", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetEvent Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetEventResultSet = function(resultStream) {
    GetEventResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    A Google Calendar Event resource.
    */
    this.getEvent = function()  {
    	return new  GoogleEvent(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(GetEvent, choreography.Choreography);
util.inherits(GetEventInputSet, choreography.InputSet);
util.inherits(GetEventResultSet, choreography.ResultSet);
exports.GetEvent = GetEvent;


/*
    GetSetting
    Retrieves a specifed setting for a Google Calendar account.
*/


var GetSetting = function(session) {
    /*
        Create a new instance of the GetSetting Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/GetSetting"
    GetSetting.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetSettingResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetSettingInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetSetting
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetSettingInputSet = function() {
    GetSettingInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the SettingID input for this Choreo. ((required, string) The unique ID for the setting to retrieve. Use the GetAllSettings Choreo to retrieve settings IDs.)
        */
        this.set_SettingID = function(value) {
            this.setInput("SettingID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetSetting Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetSettingResultSet = function(resultStream) {
    GetSettingResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    A Google Calendar Setting.
    */
    this.getSetting = function()  {
    	return new  GoogleSetting(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(GetSetting, choreography.Choreography);
util.inherits(GetSettingInputSet, choreography.InputSet);
util.inherits(GetSettingResultSet, choreography.ResultSet);
exports.GetSetting = GetSetting;


/*
    SearchCalendarsByName
    Retrieves information about a calendar including the id with a given calendar name.
*/


var SearchCalendarsByName = function(session) {
    /*
        Create a new instance of the SearchCalendarsByName Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/SearchCalendarsByName"
    SearchCalendarsByName.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new SearchCalendarsByNameResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new SearchCalendarsByNameInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the SearchCalendarsByName
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var SearchCalendarsByNameInputSet = function() {
    SearchCalendarsByNameInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarName input for this Choreo. ((required, string) The name of the calendar that you want to retrieve information for. Note that if there are multiple calendars with the same name, only the first one will be returned.)
        */
        this.set_CalendarName = function(value) {
            this.setInput("CalendarName", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the Count input for this Choreo. ((optional, integer) The maximum number of calendars to search by name. The default is 15.)
        */
        this.set_Count = function(value) {
            this.setInput("Count", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the SearchCalendarsByName Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var SearchCalendarsByNameResultSet = function(resultStream) {
    SearchCalendarsByNameResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "CalendarDescription" output from this Choreo execution. ((string) The calendar description parsed from the Google response.)
        */
        this.get_CalendarDescription = function() {
            return this.getResult("CalendarDescription");
        }
        /*
        Retrieve the value for the "CalendarId" output from this Choreo execution. ((string) The calendar id parsed from the Google response.)
        */
        this.get_CalendarId = function() {
            return this.getResult("CalendarId");
        }
        /*
        Retrieve the value for the "CalendarSummary" output from this Choreo execution. ((string) The summary or calendar name parsed from the Google response.)
        */
        this.get_CalendarSummary = function() {
            return this.getResult("CalendarSummary");
        }
        /*
        Retrieve the value for the "CalendarTimezone" output from this Choreo execution. ((string) The calendar timezone parsed from the Google response.)
        */
        this.get_CalendarTimezone = function() {
            return this.getResult("CalendarTimezone");
        }
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
}

util.inherits(SearchCalendarsByName, choreography.Choreography);
util.inherits(SearchCalendarsByNameInputSet, choreography.InputSet);
util.inherits(SearchCalendarsByNameResultSet, choreography.ResultSet);
exports.SearchCalendarsByName = SearchCalendarsByName;


/*
    SearchEvents
    Allows you to search for events using a variety of search parameters.
*/


var SearchEvents = function(session) {
    /*
        Create a new instance of the SearchEvents Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/SearchEvents"
    SearchEvents.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new SearchEventsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new SearchEventsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the SearchEvents
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var SearchEventsInputSet = function() {
    SearchEventsInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The unique ID for the calendar with the events to retrieve.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the LastModified input for this Choreo. ((optional, date) An event's last modification time (as a RFC 3339 timestamp) to filter by.)
        */
        this.set_LastModified = function(value) {
            this.setInput("LastModified", value);
        }

        /*
        Set the value of the MaxAttendees input for this Choreo. ((optional, integer) The maximum number of attendees to include in the response. If there are more than the specified number of attendees, only the participant is returned.)
        */
        this.set_MaxAttendees = function(value) {
            this.setInput("MaxAttendees", value);
        }

        /*
        Set the value of the MaxResults input for this Choreo. ((optional, integer) The maximum number of events to return on one result page.)
        */
        this.set_MaxResults = function(value) {
            this.setInput("MaxResults", value);
        }

        /*
        Set the value of the MaxTime input for this Choreo. ((optional, date) The max start time to filter by (formatted like 2012-05-22T00:47:43.000Z).)
        */
        this.set_MaxTime = function(value) {
            this.setInput("MaxTime", value);
        }

        /*
        Set the value of the MinTime input for this Choreo. ((optional, date) The minimum start time to filter by (formatted like 2012-05-22T00:47:43.000Z).)
        */
        this.set_MinTime = function(value) {
            this.setInput("MinTime", value);
        }

        /*
        Set the value of the OrderBy input for this Choreo. ((optional, string) The order of the events returned in the result. Accepted values are: "startTime" (ordered by start date/time. Must set SingleEvents to 1 to use this) or "updated" (ordered by modification date/time).)
        */
        this.set_OrderBy = function(value) {
            this.setInput("OrderBy", value);
        }

        /*
        Set the value of the PageToken input for this Choreo. ((optional, integer) Indicates which result page to return. Used for paging through results.)
        */
        this.set_PageToken = function(value) {
            this.setInput("PageToken", value);
        }

        /*
        Set the value of the Query input for this Choreo. ((optional, string) A keyword search to find events.)
        */
        this.set_Query = function(value) {
            this.setInput("Query", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the ShowDeleted input for this Choreo. ((optional, boolean) Whether to include deleted events. Set to 1 (true) to include deleted events. Defaults to 0 (false).)
        */
        this.set_ShowDeleted = function(value) {
            this.setInput("ShowDeleted", value);
        }

        /*
        Set the value of the ShowHiddenInvitations input for this Choreo. ((optional, boolean) Whether to include hidden invitations in the result. Set to 1 (true) to enable. The default is 0 (false).)
        */
        this.set_ShowHiddenInvitations = function(value) {
            this.setInput("ShowHiddenInvitations", value);
        }

        /*
        Set the value of the SingleEvent input for this Choreo. ((optional, boolean) Whether to expand recurring events into instances and only return single one-off events and instances of recurring events. Defaults to 0 (false).)
        */
        this.set_SingleEvent = function(value) {
            this.setInput("SingleEvent", value);
        }

        /*
        Set the value of the Timezone input for this Choreo. ((optional, string) The time zone used in the response (i.e. America/Los_Angeles). The default is the time zone of the calendar.)
        */
        this.set_Timezone = function(value) {
            this.setInput("Timezone", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the SearchEvents Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var SearchEventsResultSet = function(resultStream) {
    SearchEventsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get All Google Calendar Events
    */
    this.getEventList = function()  {
    	return new  GoogleEventList(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(SearchEvents, choreography.Choreography);
util.inherits(SearchEventsInputSet, choreography.InputSet);
util.inherits(SearchEventsResultSet, choreography.ResultSet);
exports.SearchEvents = SearchEvents;


/*
    UpdateCalendar
    Updates the metadata for a calendar.
*/


var UpdateCalendar = function(session) {
    /*
        Create a new instance of the UpdateCalendar Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Google/Calendar/UpdateCalendar"
    UpdateCalendar.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new UpdateCalendarResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new UpdateCalendarInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the UpdateCalendar
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var UpdateCalendarInputSet = function() {
    UpdateCalendarInputSet.super_.call(this);
        /*
        Set the value of the AccessToken input for this Choreo. ((optional, string) A valid access token retrieved during the OAuth process. This is required unless you provide the ClientID, ClientSecret, and RefreshToken to generate a new access token.)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the CalendarID input for this Choreo. ((required, string) The ID for the calendar to update.)
        */
        this.set_CalendarID = function(value) {
            this.setInput("CalendarID", value);
        }

        /*
        Set the value of the ClientID input for this Choreo. ((conditional, string) The Client ID provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientID = function(value) {
            this.setInput("ClientID", value);
        }

        /*
        Set the value of the ClientSecret input for this Choreo. ((conditional, string) The Client Secret provided by Google. Required unless providing a valid AccessToken.)
        */
        this.set_ClientSecret = function(value) {
            this.setInput("ClientSecret", value);
        }

        /*
        Set the value of the NewDescription input for this Choreo. ((optional, string) The new description for the calendar to update.)
        */
        this.set_NewDescription = function(value) {
            this.setInput("NewDescription", value);
        }

        /*
        Set the value of the NewLocation input for this Choreo. ((optional, string) The new location for the calendar to update.)
        */
        this.set_NewLocation = function(value) {
            this.setInput("NewLocation", value);
        }

        /*
        Set the value of the NewSummary input for this Choreo. ((required, string) The new summary for the calendar to update.)
        */
        this.set_NewSummary = function(value) {
            this.setInput("NewSummary", value);
        }

        /*
        Set the value of the NewTimezone input for this Choreo. ((optional, string) The new timezone for the calendar to update.)
        */
        this.set_NewTimezone = function(value) {
            this.setInput("NewTimezone", value);
        }

        /*
        Set the value of the RefreshToken input for this Choreo. ((conditional, string) An OAuth Refresh Token used to generate a new access token when the original token is expired. Required unless providing a valid AccessToken.)
        */
        this.set_RefreshToken = function(value) {
            this.setInput("RefreshToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the UpdateCalendar Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var UpdateCalendarResultSet = function(resultStream) {
    UpdateCalendarResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "NewAccessToken" output from this Choreo execution. ((string) Contains a new AccessToken when the RefreshToken is provided.)
        */
        this.get_NewAccessToken = function() {
            return this.getResult("NewAccessToken");
        }
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Google. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Represents a Google Calendar Resource.
    */
    this.getCalendar = function()  {
    	return new  GoogleCalendar(JSON.parse(this.getResult('Response')));
    }
}

util.inherits(UpdateCalendar, choreography.Choreography);
util.inherits(UpdateCalendarInputSet, choreography.InputSet);
util.inherits(UpdateCalendarResultSet, choreography.ResultSet);
exports.UpdateCalendar = UpdateCalendar;




/******************************************************************************
Begin output wrapper classes
******************************************************************************/

/**
 * Utility function, to retrieve the array-type sub-item specified by the key from the parent (array) specified by the item.
 * Returns an empty array if key is not present.
 */
function getSubArrayByKey(item, key) {
	var val = item[key];
	if(val == null) {
		val = new Array();
	}
	return val;
}



/*
 The attendees of the event.
   
*/

var GoogleAttendees = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The attendee's email address, if available. This field must be present when adding an attendee.
	*/
	this.getEmail = function()  {
		return base["email"];
	}

	/*
	The attendee's response status: "needsAction" - The attendee has not responded to the invitation, "declined" - The attendee has declined the invitation, "tentative" - The attendee has tentatively accepted the invitation, "accepted" - The attendee has accepted the invitation.
	*/
	this.getResponseStatus = function()  {
		return base["responseStatus"];
	}
}


/*
 Represents a Google Calendar
   
*/

var GoogleCalendar = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Description of the calendar.
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	ETag of the resource.
	*/
	this.getEtag = function()  {
		return base["etag"];
	}

	/*
	Identifier of the calendar.
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Type of the resource ("calendar#calendar").
	*/
	this.getKind = function()  {
		return base["kind"];
	}

	/*
	Geographic location of the calendar as free-form text
	*/
	this.getLocation = function()  {
		return base["location"];
	}

	/*
	Title of the calendar.
	*/
	this.getSummary = function()  {
		return base["summary"];
	}

	/*
	The time zone of the calendar.
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}
}


/*
 Represents a Google Calendar List Entry
   (extends GoogleCalendar)
*/

var GoogleCalendarListEntry = function(base) {
 
	GoogleCalendarListEntry.super_.call(this, base);
	this.getBase = function() {
		return base;
	}
	/*
	The effective access role that the authenticated user has on the calendar: "freeBusyReader" - Provides read access to free/busy information, "reader" - Provides read access to the calendar. Private events will appear to users with reader access, but event details will be hidden, "writer" - Provides read and write access to the calendar. Private events will appear to users with writer access, and event details will be visible, "owner" - Provides ownership of the calendar. This role has all of the permissions of the writer role with the additional ability to see and manipulate ACLs.
	*/
	this.getAccessRole = function()  {
		return base["accessRole"];
	}

	/*
	The main color of the calendar in the format '#0088aa'. This property supersedes the index-based colorId property
	*/
	this.getBackgroundColor = function()  {
		return base["backgroundColor"];
	}

	/*
	The color of the calendar. This is an ID referring to an entry in the "calendar" section of the colors definition (see the "colors" endpoint).
	*/
	this.getColorId = function()  {
		return base["colorId"];
	}

	/*
	The default reminders that the authenticated user has for this calendar.
	*/
	this.getDefaultReminders = function()  {
		var listEntries = base["defaultReminders"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  GoogleDefaultReminders(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The foreground color of the calendar in the format '#ffffff'. This property supersedes the index-based colorId property.
	*/
	this.getForegroundColor = function()  {
		return base["foregroundColor"];
	}

	/*
	Type of the resource ("calendar#calendarListEntry").
	*/
	this.getKind = function()  {
		return base["kind"];
	}

	/*
	Whether the calendar content shows up in the calendar UI.
	*/
	this.getSelected = function()  {
		return base["selected"];
	}
}


/*
 The creator of the event
   
*/

var GoogleCreator = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The creator's email address, if available.
	*/
	this.getEmail = function()  {
		return base["email"];
	}
}


/*
 Default reminders
   
*/

var GoogleDefaultReminders = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The method used by this reminder: "email" - Reminders are sent via email, "sms" - Reminders are sent via SMS, "popup" - Reminders are sent via a UI popup.
	*/
	this.getMethod = function()  {
		return base["method"];
	}

	/*
	Number of minutes before the start of the event when the reminder should trigger.
	*/
	this.getMinutes = function()  {
		return base["minutes"];
	}
}


/*
 The (exclusive) end time of the event. For a recurring event, this is the end time of the first instance.
   
*/

var GoogleEnd = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The time, as a combined date-time value (formatted according to RFC 3339). A time zone offset is required unless a time zone is explicitly specified in 'timeZone'.
	*/
	this.getDateTime = function()  {
		return base["dateTime"];
	}

	/*
	The name of the time zone in which the time is specified (e.g. "Europe/Zurich"). Optional. The default is the time zone of the calendar.
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}
}


/*
 A single event on a calendar containing information such as the title of event, start and end times, and attendees.
   
*/

var GoogleEvent = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The attendees of the event.
	*/
	this.getAttendees = function()  {
		var listEntries = base["attendees"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  GoogleAttendees(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Creation time of the event (as a RFC 3339 timestamp).
	*/
	this.getCreated = function()  {
		return base["created"];
	}

	/*
	The creator of the event.
	*/
	this.getCreator = function()  {
		return new  GoogleCreator(base["creator"]);
	}

	/*
	Description of the event.
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	The (exclusive) end time of the event. For a recurring event, this is the end time of the first instance.
	*/
	this.getEnd = function()  {
		return new  GoogleEnd(base["end"]);
	}

	/*
	ETag of the resource.
	*/
	this.getEtag = function()  {
		return base["etag"];
	}

	/*
	An absolute link to this event in the Google Calendar Web UI.
	*/
	this.getHtmlLink = function()  {
		return base["htmlLink"];
	}

	/*
	Event ID in the iCalendar format.
	*/
	this.getICalUID = function()  {
		return base["iCalUID"];
	}

	/*
	Identifier of the event.
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Type of the resource ("calendar#event").
	*/
	this.getKind = function()  {
		return base["kind"];
	}

	/*
	Geographic location of the event as free-form text.
	*/
	this.getLocation = function()  {
		return base["location"];
	}

	/*
	Whether the organizer corresponds to the calendar on which this copy of the event appears
	*/
	this.getOrganizer = function()  {
		return new  GoogleOrganizer(base["organizer"]);
	}

	/*
	List of RRULE, EXRULE, RDATE and EXDATE lines for a recurring event. This field is omitted for single events or instances of recurring events.
	*/
	this.getRecurrence = function()  {
		var listEntries = base["recurrence"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = listEntries[i];
			}
		}
		return resultArray;
	}

	/*
	If the event doesn't use the default reminders, this lists the reminders specific to the event, or, if not set, indicates that no reminders are set for this event.
	*/
	this.getReminders = function()  {
		return new  GoogleReminders(base["reminders"]);
	}

	/*
	Sequence number as per iCalendar.
	*/
	this.getSequence = function()  {
		return base["sequence"];
	}

	/*
	The (inclusive) start time of the event. For a recurring event, this is the start time of the first instance.
	*/
	this.getStart = function()  {
		return new  GoogleStart(base["start"]);
	}

	/*
	Status of the event: "confirmed" - The event is confirmed. This is the default status, "tentative" - The event is tentatively confirmed, "cancelled" - The event is cancelled.
	*/
	this.getStatus = function()  {
		return base["status"];
	}

	/*
	Title of the event.
	*/
	this.getSummary = function()  {
		return base["summary"];
	}

	/*
	Last modification time of the event (as a RFC 3339 timestamp).
	*/
	this.getUpdated = function()  {
		return base["updated"];
	}
}


/*
 A listing of Google Calendar Events
 
*/

var GoogleEventList = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The user's access role for this calendar: "none" - The user has no access, "freeBusyReader" - The user has read access to free/busy information, "reader" - The user has read access to the calendar. Private events will appear to users with reader access, but event details will be hidden, "writer" - The user has read and write access to the calendar. Private events will appear to users with writer access, and event details will be visible, "owner" - The user has ownership of the calendar. This role has all of the permissions of the writer role with the additional ability to see and manipulate ACLs.
	*/
	this.getAccessRole = function()  {
		return base["accessRole"];
	}

	/*
	*/
	this.getDefaultReminders = function()  {
		var listEntries = base["defaultReminders"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  GoogleDefaultReminders(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Description of the calendar.
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	ETag of the collection.
	*/
	this.getEtag = function()  {
		return base["etag"];
	}

	/*
	A list of Google Calendar Events.
	*/
	this.getEvents = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  GoogleEvent(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Type of the collection ("calendar#events").
	*/
	this.getKind = function()  {
		return base["kind"];
	}

	/*
	Title of the calendar
	*/
	this.getSummary = function()  {
		return base["summary"];
	}

	/*
	The time zone of the calendar.
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}

	/*
	Last modification time of the calendar (as a RFC 3339 timestamp)
	*/
	this.getUpdated = function()  {
		return base["updated"];
	}
}


/*
 The organizer of the event. If the organizer is also an attendee, this is indicated with a separate entry in 'attendees' with the 'organizer' field set to True.
   
*/

var GoogleOrganizer = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The organizer's name, if available.
	*/
	this.getDisplayName = function()  {
		return base["displayName"];
	}

	/*
	The organizer's email address, if available.
	*/
	this.getEmail = function()  {
		return base["email"];
	}

	/*
	Whether the organizer corresponds to the calendar on which this copy of the event appears.  The default is False.
	*/
	this.getSelf = function()  {
		return base["self"];
	}
}


/*
 f the event doesn't use the default reminders, this lists the reminders specific to the event, or, if not set, indicates that no reminders are set for this event.
     
*/

var GoogleOverrides = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The method used by this reminder: "email" - Reminders are sent via email, "sms" - Reminders are sent via SMS, "popup" - Reminders are sent via a UI popup.
	*/
	this.getMethod = function()  {
		return base["method"];
	}

	/*
	Number of minutes before the start of the event when the reminder should trigger.
	*/
	this.getMinutes = function()  {
		return base["minutes"];
	}
}


/*
 Information about the event's reminders for the authenticated user.
   
*/

var GoogleReminders = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	If the event doesn't use the default reminders, this lists the reminders specific to the event, or, if not set, indicates that no reminders are set for this event.
	*/
	this.getOverrides = function()  {
		var listEntries = base["overrides"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  GoogleOverrides(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Whether the default reminders of the calendar apply to the event.
	*/
	this.getUseDefault = function()  {
		return base["useDefault"];
	}
}


/*
 Represents a Google Calendar user property setting.
   
*/

var GoogleSetting = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Type of the resource ("calendar#setting").
	*/
	this.getEtag = function()  {
		return base["etag"];
	}

	/*
	Name of the user setting.
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Type of the resource ("calendar#setting").
	*/
	this.getKind = function()  {
		return base["kind"];
	}

	/*
	Value of the user setting. The format of the value depends on the ID of the setting.
	*/
	this.getValue = function()  {
		return base["value"];
	}
}


/*
 A listing of all Google Calendar Settings
 
*/

var GoogleSettingList = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	ETag of the resource.
	*/
	this.getEtag = function()  {
		return base["etag"];
	}

	/*
	A list of all Google Calendar Settings.
	*/
	this.getSettings = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  GoogleSetting(listEntries[i]);
			}
		}
		return resultArray;
	}
}


/*
 The (inclusive) start time of the event. For a recurring event, this is the start time of the first instance.
   
*/

var GoogleStart = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The time, as a combined date-time value (formatted according to RFC 3339). A time zone offset is required unless a time zone is explicitly specified in 'timeZone'.
	*/
	this.getDateTime = function()  {
		return base["dateTime"];
	}

	/*
	The name of the time zone in which the time is specified (e.g. "Europe/Zurich"). Optional. The default is the time zone of the calendar.
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}
}

 
util.inherits(GoogleCalendarListEntry, GoogleCalendar);
