// temboo/Library/Foursquare/Users.js

var util = require("util");
var choreography = require("temboo/core/choreography");


/*
    ApproveFriendRequest
    Approves a pending friend request from another user.
*/


var ApproveFriendRequest = function(session) {
    /*
        Create a new instance of the ApproveFriendRequest Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/ApproveFriendRequest"
    ApproveFriendRequest.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new ApproveFriendRequestResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new ApproveFriendRequestInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the ApproveFriendRequest
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var ApproveFriendRequestInputSet = function() {
    ApproveFriendRequestInputSet.super_.call(this);
        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The FourSquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((required, string) The user ID of a pending friend.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the ApproveFriendRequest Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var ApproveFriendRequestResultSet = function(resultStream) {
    ApproveFriendRequestResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    this.getUser = function()  {
    	return new  FoursquareUser(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response")["user"]);
    }
}

util.inherits(ApproveFriendRequest, choreography.Choreography);
util.inherits(ApproveFriendRequestInputSet, choreography.InputSet);
util.inherits(ApproveFriendRequestResultSet, choreography.ResultSet);
exports.ApproveFriendRequest = ApproveFriendRequest;


/*
    Badges
    Returns badges for a given user.
*/


var Badges = function(session) {
    /*
        Create a new instance of the Badges Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/Badges"
    Badges.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new BadgesResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new BadgesInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the Badges
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var BadgesInputSet = function() {
    BadgesInputSet.super_.call(this);
        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Identity of the user to get badges for. Defaults to "self" to get lists of the acting user.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the Badges Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var BadgesResultSet = function(resultStream) {
    BadgesResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get badge ids and badge images
    */
    this.getSets = function()  {
    	var listEntries = getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "sets")["groups"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareSet(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(Badges, choreography.Choreography);
util.inherits(BadgesInputSet, choreography.InputSet);
util.inherits(BadgesResultSet, choreography.ResultSet);
exports.Badges = Badges;


/*
    CheckinsByUser
    Retrieve a list of check-ins for an authenticated user.
*/


var CheckinsByUser = function(session) {
    /*
        Create a new instance of the CheckinsByUser Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/CheckinsByUser"
    CheckinsByUser.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new CheckinsByUserResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new CheckinsByUserInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the CheckinsByUser
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var CheckinsByUserInputSet = function() {
    CheckinsByUserInputSet.super_.call(this);
        /*
        Set the value of the AfterTimeStamp input for this Choreo. ((optional, date) Retrieve the first results after the seconds entered since epoch time.)
        */
        this.set_AfterTimeStamp = function(value) {
            this.setInput("AfterTimeStamp", value);
        }

        /*
        Set the value of the BeforeTimeStamp input for this Choreo. ((optional, date) Retrieve the first results prior to the seconds specified. Useful for paging backward in time.)
        */
        this.set_BeforeTimeStamp = function(value) {
            this.setInput("BeforeTimeStamp", value);
        }

        /*
        Set the value of the Limit input for this Choreo. ((optional, integer) The total number of results to be returned, up to 250.)
        */
        this.set_Limit = function(value) {
            this.setInput("Limit", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the Offset input for this Choreo. ((optional, integer) The number of results to skip. Used to page through results.)
        */
        this.set_Offset = function(value) {
            this.setInput("Offset", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Only 'self' is supported at this moment by the Foursquare API. Defaults to: self.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the CheckinsByUser Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var CheckinsByUserResultSet = function(resultStream) {
    CheckinsByUserResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get user checkins
    */
    this.getCheckins = function()  {
    	var listEntries = getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "checkins")["items"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareCheckin(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(CheckinsByUser, choreography.Choreography);
util.inherits(CheckinsByUserInputSet, choreography.InputSet);
util.inherits(CheckinsByUserResultSet, choreography.ResultSet);
exports.CheckinsByUser = CheckinsByUser;


/*
    DenyFriendRequest
    Denies a pending friend request from another user.
*/


var DenyFriendRequest = function(session) {
    /*
        Create a new instance of the DenyFriendRequest Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/DenyFriendRequest"
    DenyFriendRequest.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new DenyFriendRequestResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new DenyFriendRequestInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the DenyFriendRequest
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var DenyFriendRequestInputSet = function() {
    DenyFriendRequestInputSet.super_.call(this);
        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The FourSquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((required, string) The user ID of a pending friend.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the DenyFriendRequest Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var DenyFriendRequestResultSet = function(resultStream) {
    DenyFriendRequestResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    this.getUser = function()  {
    	return new  FoursquareUser(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response")["user"]);
    }
}

util.inherits(DenyFriendRequest, choreography.Choreography);
util.inherits(DenyFriendRequestInputSet, choreography.InputSet);
util.inherits(DenyFriendRequestResultSet, choreography.ResultSet);
exports.DenyFriendRequest = DenyFriendRequest;


/*
    FindUsers
    Allows a user to locate friends.
*/


var FindUsers = function(session) {
    /*
        Create a new instance of the FindUsers Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/FindUsers"
    FindUsers.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new FindUsersResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new FindUsersInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the FindUsers
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var FindUsersInputSet = function() {
    FindUsersInputSet.super_.call(this);
        /*
        Set the value of the Email input for this Choreo. ((conditional, string) A comma-delimited list of email addresses to look for. Must specify one of Name, Phone, Email, FacebookID, Twitter, or TwitterSource.)
        */
        this.set_Email = function(value) {
            this.setInput("Email", value);
        }

        /*
        Set the value of the FacebookID input for this Choreo. ((conditional, string) A comma-delimited list of Facebook ID's to look for. Must specify one of Name, Phone, Email, FacebookID, Twitter, or TwitterSource.)
        */
        this.set_FacebookID = function(value) {
            this.setInput("FacebookID", value);
        }

        /*
        Set the value of the Name input for this Choreo. ((conditional, string) A single string to search for in users' names. A single string to search for in users' names. Must specify one of Name, Phone, Email, FacebookID, Twitter, or TwitterSource.)
        */
        this.set_Name = function(value) {
            this.setInput("Name", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the Phone input for this Choreo. ((conditional, string) A comma-delimited list of phone numbers to look for. Must specify one of Name, Phone, Email, FacebookID, Twitter, or TwitterSource.)
        */
        this.set_Phone = function(value) {
            this.setInput("Phone", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the TwitterSource input for this Choreo. ((conditional, string) A single Twitter handle. Results will be users that this handle follows on Twitter who use Foursquare. Must specify one of Name, Phone, Email, FacebookID, Twitter, or TwitterSource.)
        */
        this.set_TwitterSource = function(value) {
            this.setInput("TwitterSource", value);
        }

        /*
        Set the value of the Twitter input for this Choreo. ((conditional, string) A comma-delimited list of Twitter handles to look for. Must specify one of Name, Phone, Email, FacebookID, Twitter, or TwitterSource.)
        */
        this.set_Twitter = function(value) {
            this.setInput("Twitter", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the FindUsers Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var FindUsersResultSet = function(resultStream) {
    FindUsersResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get users from the search results
    */
    this.getUser = function()  {
    	var listEntries = getSubArrayByKey(JSON.parse(this.getResult('Response')), "response")["results"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareUser(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(FindUsers, choreography.Choreography);
util.inherits(FindUsersInputSet, choreography.InputSet);
util.inherits(FindUsersResultSet, choreography.ResultSet);
exports.FindUsers = FindUsers;


/*
    Leaderboard
    Return the user's Leaderboard
*/


var Leaderboard = function(session) {
    /*
        Create a new instance of the Leaderboard Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/Leaderboard"
    Leaderboard.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new LeaderboardResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new LeaderboardInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the Leaderboard
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var LeaderboardInputSet = function() {
    LeaderboardInputSet.super_.call(this);
        /*
        Set the value of the Neighbors input for this Choreo. ((optional, integer) The number of friends' scores to return that are adjacent to your score, in ranked order.)
        */
        this.set_Neighbors = function(value) {
            this.setInput("Neighbors", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The FourSquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the Leaderboard Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var LeaderboardResultSet = function(resultStream) {
    LeaderboardResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get leaderboard items
    */
    this.getLeaderboardItems = function()  {
    	var listEntries = getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "leaderboard")["items"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareLeaderboardItem(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(Leaderboard, choreography.Choreography);
util.inherits(LeaderboardInputSet, choreography.InputSet);
util.inherits(LeaderboardResultSet, choreography.ResultSet);
exports.Leaderboard = Leaderboard;


/*
    ListFriends
    Retrieve a list of a user's friends.
*/


var ListFriends = function(session) {
    /*
        Create a new instance of the ListFriends Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/ListFriends"
    ListFriends.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new ListFriendsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new ListFriendsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the ListFriends
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var ListFriendsInputSet = function() {
    ListFriendsInputSet.super_.call(this);
        /*
        Set the value of the Limit input for this Choreo. ((optional, integer) The number of results returned, up to 500.)
        */
        this.set_Limit = function(value) {
            this.setInput("Limit", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the Offset input for this Choreo. ((optional, integer) Specify an offest to page through results.)
        */
        this.set_Offset = function(value) {
            this.setInput("Offset", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Identity of the user to get friends for. Defaults to "self" to get lists of the acting user.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the ListFriends Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var ListFriendsResultSet = function(resultStream) {
    ListFriendsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get friends
    */
    this.getUser = function()  {
    	return new  FoursquareUser(getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "friends")["items"]);
    }
}

util.inherits(ListFriends, choreography.Choreography);
util.inherits(ListFriendsInputSet, choreography.InputSet);
util.inherits(ListFriendsResultSet, choreography.ResultSet);
exports.ListFriends = ListFriends;


/*
    Mayorships
    Returns a user's mayorships.
*/


var Mayorships = function(session) {
    /*
        Create a new instance of the Mayorships Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/Mayorships"
    Mayorships.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new MayorshipsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new MayorshipsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the Mayorships
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var MayorshipsInputSet = function() {
    MayorshipsInputSet.super_.call(this);
        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Identity of the user to get mayorships for. Defaults to "self" to get lists of the acting user.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the Mayorships Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var MayorshipsResultSet = function(resultStream) {
    MayorshipsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get the venues that the user is mayor of
    */
    this.getMayorships = function()  {
    	var listEntries = getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "mayorships")["items"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareMayorship(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(Mayorships, choreography.Choreography);
util.inherits(MayorshipsInputSet, choreography.InputSet);
util.inherits(MayorshipsResultSet, choreography.ResultSet);
exports.Mayorships = Mayorships;


/*
    PendingFriendRequests
    Retrieves a list of pending friend requests for the authenticated user.
*/


var PendingFriendRequests = function(session) {
    /*
        Create a new instance of the PendingFriendRequests Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/PendingFriendRequests"
    PendingFriendRequests.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new PendingFriendRequestsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new PendingFriendRequestsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the PendingFriendRequests
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var PendingFriendRequestsInputSet = function() {
    PendingFriendRequestsInputSet.super_.call(this);
        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the PendingFriendRequests Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var PendingFriendRequestsResultSet = function(resultStream) {
    PendingFriendRequestsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get friend requests
    */
    this.getUsers = function()  {
    	var listEntries = getSubArrayByKey(JSON.parse(this.getResult('Response')), "response")["requests"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareUser(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(PendingFriendRequests, choreography.Choreography);
util.inherits(PendingFriendRequestsInputSet, choreography.InputSet);
util.inherits(PendingFriendRequestsResultSet, choreography.ResultSet);
exports.PendingFriendRequests = PendingFriendRequests;


/*
    PhotosFromUser
    Returns photos from a user.
*/


var PhotosFromUser = function(session) {
    /*
        Create a new instance of the PhotosFromUser Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/PhotosFromUser"
    PhotosFromUser.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new PhotosFromUserResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new PhotosFromUserInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the PhotosFromUser
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var PhotosFromUserInputSet = function() {
    PhotosFromUserInputSet.super_.call(this);
        /*
        Set the value of the Limit input for this Choreo. ((optional, integer) Number of results to return, up to 500.)
        */
        this.set_Limit = function(value) {
            this.setInput("Limit", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the Offset input for this Choreo. ((optional, integer) Used to page through results.)
        */
        this.set_Offset = function(value) {
            this.setInput("Offset", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Identity of the user to get badges for. Defaults to "self" to get lists of the acting user.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the PhotosFromUser Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var PhotosFromUserResultSet = function(resultStream) {
    PhotosFromUserResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get user photos
    */
    this.getPhotos = function()  {
    	var listEntries = getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "photos")["items"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquarePhoto(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(PhotosFromUser, choreography.Choreography);
util.inherits(PhotosFromUserInputSet, choreography.InputSet);
util.inherits(PhotosFromUserResultSet, choreography.ResultSet);
exports.PhotosFromUser = PhotosFromUser;


/*
    TipsFromUser
    Returns tips from a user.
*/


var TipsFromUser = function(session) {
    /*
        Create a new instance of the TipsFromUser Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/TipsFromUser"
    TipsFromUser.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new TipsFromUserResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new TipsFromUserInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the TipsFromUser
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var TipsFromUserInputSet = function() {
    TipsFromUserInputSet.super_.call(this);
        /*
        Set the value of the Latitude input for this Choreo. ((optional, decimal) The latitude point of the user's location.)
        */
        this.set_Latitude = function(value) {
            this.setInput("Latitude", value);
        }

        /*
        Set the value of the Limit input for this Choreo. ((optional, integer) Number of results to return, up to 500.)
        */
        this.set_Limit = function(value) {
            this.setInput("Limit", value);
        }

        /*
        Set the value of the Longitude input for this Choreo. ((optional, decimal) The longitude point of the user's location.)
        */
        this.set_Longitude = function(value) {
            this.setInput("Longitude", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The FourSquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the Offset input for this Choreo. ((optional, integer) Used to page through results.)
        */
        this.set_Offset = function(value) {
            this.setInput("Offset", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the Sort input for this Choreo. ((optional, string) Enter: recent, nearby, or popular. NEARBY requires geolat and geolong to be provided.)
        */
        this.set_Sort = function(value) {
            this.setInput("Sort", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Identity of the user to get tips for. Defaults to "self" to get lists of the acting user.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the TipsFromUser Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var TipsFromUserResultSet = function(resultStream) {
    TipsFromUserResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get tips for this user
    */
    this.getList = function()  {
    	return new  FoursquareList(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response")["list"]);
    }
}

util.inherits(TipsFromUser, choreography.Choreography);
util.inherits(TipsFromUserInputSet, choreography.InputSet);
util.inherits(TipsFromUserResultSet, choreography.ResultSet);
exports.TipsFromUser = TipsFromUser;


/*
    UserLists
    Retrieves user lists.
*/


var UserLists = function(session) {
    /*
        Create a new instance of the UserLists Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/UserLists"
    UserLists.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new UserListsResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new UserListsInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the UserLists
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var UserListsInputSet = function() {
    UserListsInputSet.super_.call(this);
        /*
        Set the value of the Group input for this Choreo. ((optional, string) Used to narrow down the lists to returns. Valid values are: created, edited, followed, friends, and suggested. See documentation for definitions of these parameter values.)
        */
        this.set_Group = function(value) {
            this.setInput("Group", value);
        }

        /*
        Set the value of the Latitude input for this Choreo. ((conditional, decimal) Latitude of user's location. Required in order to return the suggested group.)
        */
        this.set_Latitude = function(value) {
            this.setInput("Latitude", value);
        }

        /*
        Set the value of the Longitude input for this Choreo. ((conditional, decimal) Longitude of user's location. Required in order to return the suggested group.)
        */
        this.set_Longitude = function(value) {
            this.setInput("Longitude", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Identity of the user to get lists for. Defaults to "self" to get lists of the acting user.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the UserLists Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var UserListsResultSet = function(resultStream) {
    UserListsResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    /*
    Get list groups
    */
    this.getListGroups = function()  {
    	var listEntries = getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "lists")["groups"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareListGroup(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(UserLists, choreography.Choreography);
util.inherits(UserListsInputSet, choreography.InputSet);
util.inherits(UserListsResultSet, choreography.ResultSet);
exports.UserLists = UserLists;


/*
    VenueHistory
    Returns a list of all venues visited by the specified user, along with how many visits and when they were last there. 
*/


var VenueHistory = function(session) {
    /*
        Create a new instance of the VenueHistory Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/Foursquare/Users/VenueHistory"
    VenueHistory.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new VenueHistoryResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new VenueHistoryInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the VenueHistory
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var VenueHistoryInputSet = function() {
    VenueHistoryInputSet.super_.call(this);
        /*
        Set the value of the AfterTimeStamp input for this Choreo. ((optional, date) Retrieve the first results after the seconds entered since epoch time.)
        */
        this.set_AfterTimeStamp = function(value) {
            this.setInput("AfterTimeStamp", value);
        }

        /*
        Set the value of the BeforeTimeStamp input for this Choreo. ((optional, date) Retrieve the first results prior to the seconds specified. Useful for paging backward in time.)
        */
        this.set_BeforeTimeStamp = function(value) {
            this.setInput("BeforeTimeStamp", value);
        }

        /*
        Set the value of the CategoryID input for this Choreo. ((optional, string) Limits returned venues to those in this category. If specifying a top-level category, all sub-categories will also match the query.)
        */
        this.set_CategoryID = function(value) {
            this.setInput("CategoryID", value);
        }

        /*
        Set the value of the OauthToken input for this Choreo. ((required, string) The Foursquare API Oauth token string.)
        */
        this.set_OauthToken = function(value) {
            this.setInput("OauthToken", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that response should be in. Can be set to xml or json. Defaults to json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the UserID input for this Choreo. ((optional, string) Only 'self' is supported at this moment by the Foursquare API. Defaults to: self.)
        */
        this.set_UserID = function(value) {
            this.setInput("UserID", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the VenueHistory Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var VenueHistoryResultSet = function(resultStream) {
    VenueHistoryResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from Foursquare. Corresponds to the ResponseFormat input. Defaults to JSON.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
    /*
    Get the response status code
    */
    this.getMeta = function()  {
    	return new  FoursquareMeta(JSON.parse(this.getResult('Response'))["meta"]);
    }
    /*
    Get the count of unread notifications for the authenticated user
    */
    this.getNotifications = function()  {
    	var listEntries = JSON.parse(this.getResult('Response'))["notifications"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareNotification(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
    this.getVenues = function()  {
    	var listEntries = getSubArrayByKey(getSubArrayByKey(getSubArrayByKey(JSON.parse(this.getResult('Response')), "response"), "venues"), "items")["venue"];
    	var resultArray = new Array();
    	if(listEntries != null) {
    		for(var i=0; i < listEntries.length; i++) {
    			resultArray[i] = new  FoursquareVenue(listEntries[i]);
    		}
    	}
    	return resultArray;
    }
}

util.inherits(VenueHistory, choreography.Choreography);
util.inherits(VenueHistoryInputSet, choreography.InputSet);
util.inherits(VenueHistoryResultSet, choreography.ResultSet);
exports.VenueHistory = VenueHistory;




/******************************************************************************
Begin output wrapper classes
******************************************************************************/

/**
 * Utility function, to retrieve the array-type sub-item specified by the key from the parent (array) specified by the item.
 * Returns an empty array if key is not present.
 */
function getSubArrayByKey(item, key) {
	var val = item[key];
	if(val == null) {
		val = new Array();
	}
	return val;
}



/*
 Contains badges associated with the checkin
 
*/

var FoursquareBadge = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	A canonical ID for this badge
	*/
	this.getBadgeId = function()  {
		return base["badgeId"];
	}

	/*
	The badge description
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	A unique identifier for this badge
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get the badge image information
	*/
	this.getBadgeImage = function()  {
		return new  FoursquareBadgeImage(base["image"]);
	}

	/*
	The name of the badge
	*/
	this.getName = function()  {
		return base["name"];
	}
}


/*
 Contains the image information for badges
     
*/

var FoursquareBadgeImage = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	*/
	this.getPrefix = function()  {
		return base["prefix"];
	}

	/*
	*/
	this.getSizes = function()  {
		var listEntries = base["sizes"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = listEntries[i];
			}
		}
		return resultArray;
	}
}


/*
 Contains the count of the number of times the acting user has been here. Absent if there is no acting user 
*/

var FoursquareBeenHere = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The number of times the acting user has been at the venue
	*/
	this.getCount = function()  {
		return base["count"];
	}
}


/*
  Contains information about the categories that have been applied to this foursquare object
 
*/

var FoursquareCategory = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The icon for this category
	*/
	this.getIcon = function()  {
		return base["icon"];
	}

	/*
	The id of the category
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The name of the category
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	*/
	this.getParents = function()  {
		var listEntries = base["parents"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = listEntries[i];
			}
		}
		return resultArray;
	}

	/*
	The plural name for the category
	*/
	this.getPluralName = function()  {
		return base["pluralName"];
	}

	/*
	Whether or not this category is tagged as the primary category
	*/
	this.getPrimary = function()  {
		return base["primary"];
	}

	/*
	The short name for the category
	*/
	this.getShortName = function()  {
		return base["shortName"];
	}
}


/*
 A checkin object
 
*/

var FoursquareCheckin = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	this.getBadges = function()  {
		var listEntries = getSubArrayByKey(base, "badges")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareBadge(listEntries[i]);
			}
		}
		return resultArray;
	}

	this.getComments = function()  {
		var listEntries = getSubArrayByKey(base, "comments")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareComment(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The epoch timestamp for when the checkin was created
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The id the checkin
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Whether or not the checkin has been liked
	*/
	this.getLike = function()  {
		return base["like"];
	}

	/*
	Get likes for this checkin
	*/
	this.getLikes = function()  {
		return new  FoursquareLikes(base["likes"]);
	}

	/*
	Get information about overlapping checkins
	*/
	this.getOverlaps = function()  {
		var listEntries = getSubArrayByKey(base, "overlaps")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareOverlap(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get photos for this checkin
	*/
	this.getPhotos = function()  {
		var listEntries = getSubArrayByKey(base, "photos")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePhoto(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get the posts for this checkin. May not be present.
	*/
	this.getPosts = function()  {
		var listEntries = getSubArrayByKey(base, "posts")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePost(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get scores for this checkin
	*/
	this.getScores = function()  {
		var listEntries = getSubArrayByKey(base, "score")["scores"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareScore(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The message from check-in
	*/
	this.getShout = function()  {
		return base["shout"];
	}

	/*
	Get the source information for the app that created the checkin
	*/
	this.getSource = function()  {
		return new  FoursquareSource(base["source"]);
	}

	/*
	The timezone associated with this checkin
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}

	/*
	A string representation of the time zone where this check-in occurred
	*/
	this.getTimeZoneOffset = function()  {
		return base["timeZoneOffset"];
	}

	/*
	The type of foursquare object. One of checkin, shout, or venueless.
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	Get the user associated with this checkin
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}

	/*
	Get the venue for this checkin
	*/
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}
}


/*
 Contains the collaborators for this item
 
*/

var FoursquareCollaborator = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Contains the user objects that are collaborators
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["users"]);
	}
}


/*
 A list of comments objects
 
*/

var FoursquareComment = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	*/
	this.getText = function()  {
		return base["text"];
	}

	/*
	Get the user that left this comment
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 An object containing twitter, phone, and formattedPhone of the user or venue contact. All can be present, and all can be absent from the response.
 
*/

var FoursquareContact = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The email address of the contact
	*/
	this.getEmail = function()  {
		return base["email"];
	}

	/*
	The facebook id of the contact
	*/
	this.getFacebook = function()  {
		return base["facebook"];
	}

	/*
	The formatted phone number of the contact
	*/
	this.getFormattedPhone = function()  {
		return base["formattedPhone"];
	}

	/*
	The unformatted phone number of the contact
	*/
	this.getPhone = function()  {
		return base["phone"];
	}

	/*
	The twitter handle of the contact
	*/
	this.getTwitter = function()  {
		return base["twitter"];
	}
}


/*
 Contains a count of tips that are done
 
*/

var FoursquareDone = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The number of tips that are done
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Get groups of people that have marked this tip done
	*/
	this.getPeopleGroup = function()  {
		var listEntries = base["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePeopleGroup(listEntries[i]);
			}
		}
		return resultArray;
	}
}


/*
 Contains details about venue events
   
*/

var FoursquareEvent = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Whether or not this is an all day event
	*/
	this.getAllDay = function()  {
		return base["allDay"];
	}

	/*
	Get categories for this event
	*/
	this.getCategories = function()  {
		var listEntries = base["categories"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareCategory(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The date of the even as an epoch timestamp
	*/
	this.getDate = function()  {
		return base["date"];
	}

	/*
	Get foreign ids
	*/
	this.getForeignIds = function()  {
		var listEntries = getSubArrayByKey(base, "foreignIds")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareForeignIds(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get the people at this event now
	*/
	this.getHereNow = function()  {
		return new  FoursquareHereNow(base["hereNow"]);
	}

	/*
	The event id
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The name of the event
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	Get stats for the event
	*/
	this.getStats = function()  {
		return new  FoursquareStats(base["stats"]);
	}

	/*
	The event timezone
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}

	/*
	The url for the event
	*/
	this.getUrl = function()  {
		return base["url"];
	}
}


/*
 An object containing the count of followers for the item
 
*/

var FoursquareFollowers = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The number of followers for this todo
	*/
	this.getCount = function()  {
		return base["count"];
	}
}


/*
 Contains the domain and id of the third party provider
     
*/

var FoursquareForeignIds = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The domain of the 3rd party provider
	*/
	this.getDomain = function()  {
		return base["domain"];
	}

	/*
	The id of the 3rd party provider
	*/
	this.getId = function()  {
		return base["id"];
	}
}


/*
 Contains information about who is here now
 
*/

var FoursquareHereNow = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The number of people here now
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Contains friends and others as types
	*/
	this.getPeopleGroups = function()  {
		var listEntries = base["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePeopleGroup(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	A summary of people here now
	*/
	this.getSummary = function()  {
		return base["summary"];
	}
}


/*

     
*/

var FoursquareHereNowItem = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	When the checkin was created in epoch format
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The id of the checkin
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Whether or not this checkin is from the mayor
	*/
	this.getIsMayor = function()  {
		return base["isMayor"];
	}

	/*
	The timezone associated with the checkin
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}

	/*
	The timezone offset associated with the checkin
	*/
	this.getTimeZoneOffset = function()  {
		return base["timeZoneOffset"];
	}

	/*
	The type of checkin
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	Get the user details
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 Contains a count and list of people at this venue right now
   
*/

var FoursquareHereNowList = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The total number of people here now
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Get checkins and user info for people here now
	*/
	this.getHereNowItems = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareHereNowItem(listEntries[i]);
			}
		}
		return resultArray;
	}
}


/*
 Contains the hours during the week that the venue is open along with any named hours segments in a human-readable format
   
*/

var FoursquareHours = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Whether or not the venue is open
	*/
	this.getIsOpen = function()  {
		return base["isOpen"];
	}

	/*
	The venue status
	*/
	this.getStatus = function()  {
		return base["status"];
	}

	/*
	Get the timeframe information for the venue including days of operation
	*/
	this.getTimeframes = function()  {
		var listEntries = base["timeframes"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareTimeframe(listEntries[i]);
			}
		}
		return resultArray;
	}
}


/*

         
*/

var FoursquareImage = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The name of the image used to contruct the complete image url
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The beginning portion of the image url used to contruct the complete image url
	*/
	this.getPrefix = function()  {
		return base["prefix"];
	}

	/*
	The available image sizes used to construct the complete image url
	*/
	this.getSizes = function()  {
		var listEntries = base["sizes"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = listEntries[i];
			}
		}
		return resultArray;
	}
}


/*
 Contains information about insights related to this checkin
     
*/

var FoursquareInsight = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The url for the appropriate foursquare icon
	*/
	this.getImage = function()  {
		return base["image"];
	}

	/*
	Get the points information for the insight
	*/
	this.getPoints = function()  {
		return new  FoursquarePoints(base["points"]);
	}

	/*
	The title of the insight
	*/
	this.getTitle = function()  {
		return base["title"];
	}

	/*
	The type of insight
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 A list item object
 
*/

var FoursquareItem = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Whether or not this item has been done
	*/
	this.getDone = function()  {
		return base["done"];
	}

	/*
	The id of the item
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get lists for the authenticated user
	*/
	this.getListed = function()  {
		return new  FoursquareListed(base["listed"]);
	}

	/*
	Get the tip for this item
	*/
	this.getTip = function()  {
		return new  FoursquareTip(base["tip"]);
	}

	/*
	Whether or not this item is a todo
	*/
	this.getTodo = function()  {
		return base["todo"];
	}

	/*
	Get the venue that this item is for
	*/
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}

	/*
	How many times the venue is has been visited
	*/
	this.getVisitedCount = function()  {
		return base["visitedCount"];
	}
}


/*
 Contains a list of words that are suggested for the user to input as refinements
     
*/

var FoursquareKeywords = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The display name of the keyword
	*/
	this.getDisplayName = function()  {
		return base["displayName"];
	}

	/*
	The suggested keyword
	*/
	this.getKeyword = function()  {
		return base["keyword"];
	}
}


/*
 A leaderboard object
     
*/

var FoursquareLeaderboard = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The leaderboard rank
	*/
	this.getRank = function()  {
		return base["rank"];
	}

	/*
	Get leaderboard scores
	*/
	this.getLeaderboardScores = function()  {
		return new  FoursquareLeaderboardScores(base["scores"]);
	}

	/*
	Get the user for this leaderboard
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 Contains leaderboard items for the authenticated user
   
*/

var FoursquareLeaderboardItem = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getRank = function()  {
		return base["rank"];
	}

	/*
	Get statitics related to the leaderboard score
	*/
	this.getLeaderboardScores = function()  {
		return new  FoursquareLeaderboardScores(base["scores"]);
	}

	/*
	Get the user on this leaderboard
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 Cotains leaderboard scores for the authenticated user
       
*/

var FoursquareLeaderboardScores = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The user's total checkin count over the past 7 days
	*/
	this.getCheckinsCount = function()  {
		return base["checkinsCount"];
	}

	/*
	If the user has never scored more than a certain number of points, then this field is returned (current value is 50)
	*/
	this.getGoal = function()  {
		return base["goal"];
	}

	/*
	The highest 7-day score the user has ever achieved
	*/
	this.getMax = function()  {
		return base["max"];
	}

	/*
	The user's total score over the past 7 days
	*/
	this.getRecent = function()  {
		return base["recent"];
	}
}


/*
 Contains a count and group of people who liked an object
 
*/

var FoursquareLikes = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The count of users liked the object
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Contains groups of friends and others that have liked an object
	*/
	this.getPeopleGroups = function()  {
		var listEntries = base["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePeopleGroup(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The summary of likes
	*/
	this.getSummary = function()  {
		return base["summary"];
	}
}


/*
 A list object
 
*/

var FoursquareList = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The canonical URL for the list
	*/
	this.getCanonicalUrl = function()  {
		return base["canonicalUrl"];
	}

	/*
	Get categories for the list
	*/
	this.getCategories = function()  {
		var listEntries = getSubArrayByKey(base, "categories")["item"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareCategory(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Whether or not the list has collaborators
	*/
	this.getCollaborative = function()  {
		return base["collaborative"];
	}

	/*
	Get collaborators for this list
	*/
	this.getCollaborators = function()  {
		var listEntries = getSubArrayByKey(base, "collaborators")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareCollaborator(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The description of the list
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	The count of items that are done
	*/
	this.getDoneCount = function()  {
		return base["doneCount"];
	}

	/*
	Whether or not the list is editable
	*/
	this.getEditable = function()  {
		return base["editable"];
	}

	/*
	Get the count of followers
	*/
	this.getFollowers = function()  {
		return new  FoursquareFollowers(base["followers"]);
	}

	/*
	Whether or not there are followers of the list
	*/
	this.getFollowing = function()  {
		return base["following"];
	}

	/*
	The id of the list
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get the list items
	*/
	this.getItems = function()  {
		var listEntries = getSubArrayByKey(base, "listItems")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareItem(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The name of the list
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	Whether not the list is editable by others
	*/
	this.getPublic = function()  {
		return base["public"];
	}

	/*
	The type of list
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	The list url
	*/
	this.getUrl = function()  {
		return base["url"];
	}

	/*
	Get the user who created this list
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}

	/*
	The venue count for this list
	*/
	this.getVenueCount = function()  {
		return base["venueCount"];
	}

	/*
	The count of visits
	*/
	this.getVisitedCount = function()  {
		return base["visitedCount"];
	}
}


/*
 Contains the followers of the list
   
*/

var FoursquareListFollower = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The canonical URL for the user following the list
	*/
	this.getCanonicalUrl = function()  {
		return base["canonicalUrl"];
	}

	/*
	The first name of the user following the list
	*/
	this.getFirstName = function()  {
		return base["firstName"];
	}

	/*
	The gender of the user who is following the list
	*/
	this.getGender = function()  {
		return base["gender"];
	}

	/*
	The home city of the user who is following the list
	*/
	this.getHomeCity = function()  {
		return base["homeCity"];
	}

	/*
	The id of the user who is following the list
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The last name of the user who is following the list
	*/
	this.getLastName = function()  {
		return base["lastName"];
	}

	/*
	The url to the photo of the follower
	*/
	this.getPhoto = function()  {
		return base["photo"];
	}

	this.getTipsCount = function()  {
		return new  FoursquareTipsCount(base["tips"]);
	}
}


/*
 Contains groups of list including created, edited, followed, friends, or other
 
*/

var FoursquareListGroup = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The count of groups returned
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Get lists
	*/
	this.getLists = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareList(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The type of group
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Contains a count of lists that contain this venue
 
*/

var FoursquareListed = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	A count of lists
	*/
	this.getCount = function()  {
		return base["count"];
	}

	this.getListedItems = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareListedItems(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	A summary field for the listed items
	*/
	this.getSummary = function()  {
		return base["summary"];
	}
}


/*
 Contains list objects for the authenitcated user
   
*/

var FoursquareListedItems = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The canonical url for the list
	*/
	this.getCanonicalUrl = function()  {
		return base["canonicalUrl"];
	}

	/*
	Whether or not this list is editable by the owner's friends
	*/
	this.getCollaborative = function()  {
		return base["collaborative"];
	}

	/*
	The epoch timestamp for when this list was created
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The description of the list
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	Whether or not this item is editable
	*/
	this.getEditable = function()  {
		return base["editable"];
	}

	/*
	Get the follower count for this list item
	*/
	this.getFollowers = function()  {
		return new  FoursquareFollowers(base["followers"]);
	}

	/*
	The id of this item
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get the list items
	*/
	this.getItems = function()  {
		var listEntries = getSubArrayByKey(base, "listItems")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareItem(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The name of the list
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	Whether or not the list is public
	*/
	this.getPublic = function()  {
		return base["public"];
	}

	/*
	An epoch timestamp representing the last update time for the list
	*/
	this.getUpdatedAt = function()  {
		return base["updatedAt"];
	}

	/*
	The url for the list
	*/
	this.getUrl = function()  {
		return base["url"];
	}

	/*
	Get the user who created the list
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 Contains location information for the venue
 
*/

var FoursquareLocation = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The venue address
	*/
	this.getAddress = function()  {
		return base["address"];
	}

	/*
	The country abbreviation field for the venue address
	*/
	this.getCc = function()  {
		return base["cc"];
	}

	/*
	The city that the venue is in
	*/
	this.getCity = function()  {
		return base["city"];
	}

	/*
	The country field for the venue address
	*/
	this.getCountry = function()  {
		return base["country"];
	}

	/*
	The cross street of the venue
	*/
	this.getCrossStreet = function()  {
		return base["crossStreet"];
	}

	/*
	The latitude coordinate for the venue
	*/
	this.getLat = function()  {
		return base["lat"];
	}

	/*
	The longitude coordinate for the venue
	*/
	this.getLng = function()  {
		return base["lng"];
	}

	/*
	The postal code of the venue
	*/
	this.getPostalCode = function()  {
		return base["postalCode"];
	}

	/*
	The state that the venue is in
	*/
	this.getState = function()  {
		return base["state"];
	}
}


/*
 An object containing information about the mayor of a venue
 
*/

var FoursquareMayor = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Contains the user information for the venue mayor
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 Contains venues that this user is mayor of
   
*/

var FoursquareMayorship = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}
}


/*
 An object containing url and mobileUrl that display the menu information for this venue
 
*/

var FoursquareMenu = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The mobile url of the menu
	*/
	this.getMobileUrl = function()  {
		return base["mobileUrl"];
	}

	/*
	The type of menu
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	The url for the menu
	*/
	this.getUrl = function()  {
		return base["url"];
	}
}


/*
 An object containing the status code from Foursquare
 
*/

var FoursquareMeta = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The response status code from Foursquare
	*/
	this.getCode = function()  {
		return base["code"];
	}
}


/*
 Contains notifications of different types including insights, notificationTray, mayorship, score, message, leaderboard, etc.
 
*/

var FoursquareNotification = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Get an individual notification item
	*/
	this.getNotificationItem = function()  {
		return new  FoursquareNotificationItem(base["item"]);
	}

	/*
	The type of notification
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Contains
   
*/

var FoursquareNotificationItem = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The number of checkins
	*/
	this.getCheckins = function()  {
		return base["checkins"];
	}

	/*
	The url for the user image
	*/
	this.getImage = function()  {
		return base["image"];
	}

	/*
	Get insights
	*/
	this.getInsights = function()  {
		var listEntries = getSubArrayByKey(base, "insights")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareInsight(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get the scores for this checkin
	*/
	this.getScores = function()  {
		var listEntries = getSubArrayByKey(base, "item")["scores"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareScore(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get the leadboard object for the authenticated user
	*/
	this.getLeaderboard = function()  {
		var listEntries = base["leaderboard"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareLeaderboard(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The notification message
	*/
	this.getMessage = function()  {
		return base["message"];
	}

	/*
	The total number of items
	*/
	this.getTotal = function()  {
		return base["total"];
	}

	/*
	The type of notification item
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	The number of unread notifications
	*/
	this.getUnreadCount = function()  {
		return base["unreadCount"];
	}

	/*
	Get the user information for this notification item
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 The segments of the days in this timeframe in which the venue is open
       
*/

var FoursquareOpen = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The time the venue closes in HHMM (24hr) format
	*/
	this.getEnd = function()  {
		return base["end"];
	}

	/*
	The human readable version the open and close times
	*/
	this.getRenderedTime = function()  {
		return base["renderedTime"];
	}

	/*
	The time the venue opens in HHMM (24hr) format
	*/
	this.getStart = function()  {
		return base["start"];
	}
}


/*
 Contains information about overlapping checkins
   
*/

var FoursquareOverlap = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The epoch timestamp for the overlapping checkin
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The id of the overlapping ehckin
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The timezone for the overlapping checkin
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}

	/*
	The timezone offset for the checkin
	*/
	this.getTimeZoneOffset = function()  {
		return base["timeZoneOffset"];
	}

	/*
	The type of foursquare object
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	Get the users overlapping with the checkin
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 An object containing the count of page updates
   
*/

var FoursquarePageUpdates = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The count of updates
	*/
	this.getCount = function()  {
		return base["count"];
	}
}


/*
 Contains friends and others as types
 
*/

var FoursquarePeopleGroup = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The number of people here
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Get users in this group
	*/
	this.getUsers = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareUser(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The name of the group
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The group type
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Contains a photo object
 
*/

var FoursquarePhoto = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The height of the photo in pixels
	*/
	this.getHeight = function()  {
		return base["height"];
	}

	/*
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The start of the URL for this photo
	*/
	this.getPrefix = function()  {
		return base["prefix"];
	}

	this.getSizes = function()  {
		var listEntries = getSubArrayByKey(base, "sizes")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareSize(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get the source information for this photo
	*/
	this.getSource = function()  {
		return new  FoursquareSource(base["source"]);
	}

	/*
	The end of the URL for this photo
	*/
	this.getSuffix = function()  {
		return base["suffix"];
	}

	/*
	Get the tip for this photo
	*/
	this.getTip = function()  {
		return new  FoursquareTip(base["tip"]);
	}

	/*
	*/
	this.getUrl = function()  {
		return base["url"];
	}

	/*
	Get the user that took this photo
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}

	/*
	Get the venue associated with this photo
	*/
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}

	/*
	*/
	this.getVisibility = function()  {
		return base["visibility"];
	}

	/*
	The width of this photo in pixels
	*/
	this.getWidth = function()  {
		return base["width"];
	}
}


/*
 Contains checkin photos and venue photos
 
*/

var FoursquarePhotoGroup = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Get a photo associated with the venue
	*/
	this.getPhotos = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePhoto(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The name of the photo group
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The photo group type
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Contains information for points for the insight
       
*/

var FoursquarePoints = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	this.getImage = function()  {
		return new  FoursquareImage(base["image"]);
	}

	/*
	A foursquare generated message about the checkin
	*/
	this.getMessage = function()  {
		return base["message"];
	}

	/*
	The number of points
	*/
	this.getPoints = function()  {
		return base["points"];
	}
}


/*
 Contains information about the checkin post including the id, text, timestamp, and source information
 
*/

var FoursquarePost = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The timestamp that the post was created in epoch format
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The id of the post created
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get the source for this post
	*/
	this.getSource = function()  {
		return new  FoursquareSource(base["source"]);
	}

	/*
	The text of the post
	*/
	this.getText = function()  {
		return base["text"];
	}
}


/*
 Contains price information for this venue
   
*/

var FoursquarePrice = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The message for pricing
	*/
	this.getMessage = function()  {
		return base["message"];
	}

	/*
	The pricing tier
	*/
	this.getTier = function()  {
		return base["tier"];
	}
}


/*
 Contains information about why this place may be of interest to the acting user
         
*/

var FoursquareReason = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The message text for the reason
	*/
	this.getMessage = function()  {
		return base["message"];
	}

	/*
	The reason type
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Contains recent checkins for the authenticated user
   
*/

var FoursquareRecent = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The epoch timestamp for when the checkin created
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The checkin id
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get photos for recent checkins
	*/
	this.getPhotos = function()  {
		var listEntries = getSubArrayByKey(base, "photos")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePhoto(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get the source information for this checkin
	*/
	this.getSource = function()  {
		return new  FoursquareSource(base["source"]);
	}

	/*
	The timezone associated with this checkin
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}

	/*
	String representation of the time zone where this check-in occurred
	*/
	this.getTimeZoneOffset = function()  {
		return base["timeZoneOffset"];
	}

	/*
	The type of checkin. One of checkin, shout, or venueless.
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	Get the user for this checkin
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}

	/*
	Get the venue that was checked into
	*/
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}
}


/*

     
*/

var FoursquareRecommendationGroup = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	this.getRecommendationItem = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareRecommendationItem(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The name of the group
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The group type
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*

       
*/

var FoursquareRecommendationItem = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Get reasons why this place may be of interest to the acting user
	*/
	this.getReasons = function()  {
		var listEntries = getSubArrayByKey(base, "reasons")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareReason(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get suggested tips
	*/
	this.getTips = function()  {
		var listEntries = base["tips"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareTip(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get suggested venues
	*/
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}
}


/*
 
*/

var FoursquareRecommendations = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	this.getRecommendationGroups = function()  {
		var listEntries = base["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareRecommendationGroup(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	A full text name for the location the user searched
	*/
	this.getHeaderFullLocation = function()  {
		return base["headerFullLocation"];
	}

	/*
	A text name for the location the user searched
	*/
	this.getHeaderLocation = function()  {
		return base["headerLocation"];
	}

	/*
	More granular location information
	*/
	this.getHeaderLocationGranularity = function()  {
		return base["headerLocationGranularity"];
	}

	/*
	A message to the user based on their current context
	*/
	this.getHeaderMessage = function()  {
		return base["headerMessage"];
	}

	/*
	Get keyword suggestions
	*/
	this.getKeywords = function()  {
		return new  FoursquareKeywords(getSubArrayByKey(base, "keywords")["items"]);
	}

	/*
	When no radius is specified, this represents the radius that was used for the query (based upon the density of venues in the query area)
	*/
	this.getSuggestedRadius = function()  {
		return base["suggestedRadius"];
	}

	/*
	Get warnings about the request
	*/
	this.getWarning = function()  {
		return new  FoursquareWarning(base["warning"]);
	}
}


/*
 Contains a reply object
 
*/

var FoursquareReply = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The epoch timestamp for the reply
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	The reply id
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get the source that created this reply
	*/
	this.getSource = function()  {
		return new  FoursquareSource(base["source"]);
	}

	/*
	The text of the reply
	*/
	this.getText = function()  {
		return base["text"];
	}

	/*
	The url the reply
	*/
	this.getUrl = function()  {
		return base["url"];
	}
}


/*
 Contains the scores for this checkin
 
*/

var FoursquareScore = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getIcon = function()  {
		return base["icon"];
	}

	/*
	*/
	this.getMessage = function()  {
		return base["message"];
	}

	/*
	*/
	this.getPoints = function()  {
		return base["points"];
	}
}


/*
 Contains the named segments of the days in this timeframe in which the venue is open
       
*/

var FoursquareSegment = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The time as HHMM (24hr) at which the segment ends
	*/
	this.getEnd = function()  {
		return base["end"];
	}

	/*
	The time as HHMM (24hr) at which the segment begins
	*/
	this.getStart = function()  {
		return base["start"];
	}
}


/*
 Contains badge IDs and badge image information
   
*/

var FoursquareSet = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Get badge image info
	*/
	this.getBadgeImage = function()  {
		return new  FoursquareBadgeImage(base["image"]);
	}

	/*
	Contains badge ids
	*/
	this.getBadgeIDs = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = listEntries[i];
			}
		}
		return resultArray;
	}

	/*
	The name of the badge
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The badge type
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Available sizes for an image
 
*/

var FoursquareSize = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The height of an image
	*/
	this.getHeight = function()  {
		return base["height"];
	}

	/*
	The url for an image
	*/
	this.getUrl = function()  {
		return base["url"];
	}

	/*
	The width of the image
	*/
	this.getWidth = function()  {
		return base["width"];
	}
}


/*
 An object containing the source information about this object
 
*/

var FoursquareSource = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The detail url for the application that created the object
	*/
	this.getDetailUrl = function()  {
		return base["detailUrl"];
	}

	/*
	The icon for the application that create the object
	*/
	this.getIcon = function()  {
		return base["icon"];
	}

	/*
	The id of the application that created the object
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The name of the application that created the object
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The link to the photo of the application that created the object
	*/
	this.getPhoto = function()  {
		return base["photo"];
	}

	/*
	The url for the application that created the object
	*/
	this.getUrl = function()  {
		return base["url"];
	}
}


/*
 Contains specials offered by a venue
 
*/

var FoursquareSpecial = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	*/
	this.getFinePrint = function()  {
		return base["finePrint"];
	}

	/*
	*/
	this.getIcon = function()  {
		return base["icon"];
	}

	/*
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	*/
	this.getMessage = function()  {
		return base["message"];
	}

	/*
	*/
	this.getProvider = function()  {
		return base["provider"];
	}

	/*
	*/
	this.getRedemption = function()  {
		return base["redemption"];
	}

	/*
	*/
	this.getTitle = function()  {
		return base["title"];
	}

	/*
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Contains a list of specials
   
*/

var FoursquareSpecialsList = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The special description
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	The fine print of the special
	*/
	this.getFinePrint = function()  {
		return base["finePrint"];
	}

	/*
	The icon name
	*/
	this.getIcon = function()  {
		return base["icon"];
	}

	/*
	The special id
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get the likes for the special
	*/
	this.getLikes = function()  {
		return new  FoursquareLikes(base["likes"]);
	}

	/*
	The message text for this special
	*/
	this.getMessage = function()  {
		return base["message"];
	}

	/*
	The provider of the special
	*/
	this.getProvider = function()  {
		return base["provider"];
	}

	/*
	The redemption type
	*/
	this.getRedemption = function()  {
		return base["redemption"];
	}

	/*
	The title of the special
	*/
	this.getTitle = function()  {
		return base["title"];
	}

	/*
	The special type
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	Get the venue offering the special
	*/
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}
}


/*
 Contains statistics for the venue including checkinsCount, tipCount, and userCount
 
*/

var FoursquareStats = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The total number of checkins for this venue
	*/
	this.getCheckinsCount = function()  {
		return base["checkinsCount"];
	}

	/*
	The number of tips for this venue
	*/
	this.getTipCount = function()  {
		return base["tipCount"];
	}

	/*
	The total number of users that have checked into this venue
	*/
	this.getUsersCount = function()  {
		return base["usersCount"];
	}
}


/*
 An object containing the days
     
*/

var FoursquareTimeframe = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Days represented as integers with Monday = 1,..., Sunday = 7 on which these hours apply
	*/
	this.getDays = function()  {
		return base["days"];
	}

	/*
	Indicates if this timeframe includes today
	*/
	this.getIncludesToday = function()  {
		return base["includesToday"];
	}

	/*
	Get the start, end, and rendered time for the venue
	*/
	this.getOpen = function()  {
		var listEntries = base["open"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareOpen(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get segment information for the venue
	*/
	this.getSegments = function()  {
		var listEntries = base["segments"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareSegment(listEntries[i]);
			}
		}
		return resultArray;
	}
}


/*
 Contains information about tips associated with a venue
 
*/

var FoursquareTip = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The url for the tip
	*/
	this.getCanonicalUrl = function()  {
		return base["canonicalUrl"];
	}

	/*
	The date that this tip was created in epoch format
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	Get the count of tips that have been done
	*/
	this.getDone = function()  {
		return new  FoursquareDone(base["done"]);
	}

	/*
	The id of the tip
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Get the photos for this tip
	*/
	this.getPhoto = function()  {
		return new  FoursquarePhoto(getSubArrayByKey(base, "items")["photo"]);
	}

	/*
	Get the likes for this tip
	*/
	this.getLikes = function()  {
		return new  FoursquareLikes(base["likes"]);
	}

	/*
	The url to the photo
	*/
	this.getPhotourl = function()  {
		return base["photourl"];
	}

	/*
	The text of the tip
	*/
	this.getText = function()  {
		return base["text"];
	}

	/*
	Get to count of todos for this tip
	*/
	this.getTodo = function()  {
		return new  FoursquareTodo(base["todo"]);
	}

	/*
	The tip url
	*/
	this.getUrl = function()  {
		return base["url"];
	}

	/*
	Get the user that created this tip
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}
}


/*
 Contains tip details including users that have marked a tip done and users that have marked a tip as a todo
 
*/

var FoursquareTipDetails = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The url for the tip
	*/
	this.getCanonicalUrl = function()  {
		return base["canonicalUrl"];
	}

	/*
	An epoch timestamp when the tip was created
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	Get the info for user who have marked this tip done
	*/
	this.getDone = function()  {
		return new  FoursquareDone(base["done"]);
	}

	/*
	The id of this tip
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	Whether or not the tip has been liked
	*/
	this.getLike = function()  {
		return base["like"];
	}

	/*
	Get the likes for this tip
	*/
	this.getLikes = function()  {
		return new  FoursquareLikes(base["likes"]);
	}

	/*
	Get the the count of tip
	*/
	this.getListed = function()  {
		return new  FoursquareListed(base["listed"]);
	}

	/*
	The actual tip
	*/
	this.getText = function()  {
		return base["text"];
	}

	/*
	Get information about people that have indicated this tip as a todo
	*/
	this.getTodo = function()  {
		return new  FoursquareTodo(base["todo"]);
	}

	/*
	Get the user that created this tip
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}

	/*
	Get the venue associated with this tip
	*/
	this.getVenue = function()  {
		return new  FoursquareVenue(base["venue"]);
	}
}


/*
 Contains tips from others and friends
   
*/

var FoursquareTipGroup = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Get tips for the venue
	*/
	this.getTips = function()  {
		var listEntries = base["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareTip(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	The name of the tip group
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The type of tip
	*/
	this.getType = function()  {
		return base["type"];
	}
}


/*
 Contains the total count of tips for this user
 
*/

var FoursquareTipsCount = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getCount = function()  {
		return base["count"];
	}
}


/*
 An object containing the count of ToDos for this tip
 
*/

var FoursquareTodo = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	*/
	this.getCount = function()  {
		return base["count"];
	}

	/*
	Get the groups of people that want to do this tip
	*/
	this.getPeopleGroup = function()  {
		var listEntries = base["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePeopleGroup(listEntries[i]);
			}
		}
		return resultArray;
	}
}


/*
 Contains categories of todos associated with the venue
     
*/

var FoursquareTodoCategory = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The icon for the todo category
	*/
	this.getIcon = function()  {
		return base["icon"];
	}

	/*
	The id of the todo category
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The name of the todo category
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	The plural name of the todo category
	*/
	this.getPluralName = function()  {
		return base["pluralName"];
	}

	/*
	The short name of the todo category
	*/
	this.getShortName = function()  {
		return base["shortName"];
	}
}


/*
 Contains information about a todo list for the venue
   
*/

var FoursquareTodos = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The canonical url for the todo
	*/
	this.getCanonicalUrl = function()  {
		return base["canonicalUrl"];
	}

	this.getTodoCategories = function()  {
		var listEntries = getSubArrayByKey(getSubArrayByKey(base, "categories"), "items")["category"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareTodoCategory(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Whether or not other users have edited this todo
	*/
	this.getCollaborative = function()  {
		return base["collaborative"];
	}

	/*
	Get collaborators for the todo item
	*/
	this.getCollaborators = function()  {
		var listEntries = getSubArrayByKey(base, "collaborators")["items"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareCollaborator(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	A description of the todo
	*/
	this.getDescription = function()  {
		return base["description"];
	}

	/*
	The count of todos marked done
	*/
	this.getDoneCount = function()  {
		return base["doneCount"];
	}

	/*
	Whether or not this todo is editable
	*/
	this.getEditable = function()  {
		return base["editable"];
	}

	/*
	Get the number of followers for this to do
	*/
	this.getFollowers = function()  {
		return new  FoursquareFollowers(base["followers"]);
	}

	/*
	Whether or not the acting user is following the list
	*/
	this.getFollowing = function()  {
		return base["following"];
	}

	/*
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The name of the todo item
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	Whether or not this todo is public
	*/
	this.getPublic = function()  {
		return base["public"];
	}

	/*
	The type of todo
	*/
	this.getType = function()  {
		return base["type"];
	}

	/*
	The url of the todo
	*/
	this.getUrl = function()  {
		return base["url"];
	}

	/*
	The user that created the todo for this venue
	*/
	this.getUser = function()  {
		return new  FoursquareUser(base["user"]);
	}

	/*
	The venue count for the todo
	*/
	this.getVenueCount = function()  {
		return base["venueCount"];
	}

	/*
	The count of visits
	*/
	this.getVisitedCount = function()  {
		return base["visitedCount"];
	}
}


/*
 A user object
 
*/

var FoursquareUser = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The bio of the user
	*/
	this.getBio = function()  {
		return base["bio"];
	}

	/*
	An object containing twitter, phone, and formattedPhone of the user
	*/
	this.getContact = function()  {
		return new  FoursquareContact(base["contact"]);
	}

	/*
	*/
	this.getFirstName = function()  {
		return base["firstName"];
	}

	/*
	*/
	this.getGender = function()  {
		return base["gender"];
	}

	/*
	*/
	this.getHomeCity = function()  {
		return base["homeCity"];
	}

	/*
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	*/
	this.getLastName = function()  {
		return base["lastName"];
	}

	this.getListGroup = function()  {
		var listEntries = getSubArrayByKey(base, "lists")["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareListGroup(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	*/
	this.getPhoto = function()  {
		return base["photo"];
	}

	/*
	Contains the total count of tips and groups with friends and others
	*/
	this.getTipsCount = function()  {
		return new  FoursquareTipsCount(base["tips"]);
	}
}


/*
 An containing information about foursquare venue
 
*/

var FoursquareVenue = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	Contains count of the number of times the acting user has been here. Absent if there is no acting user
	*/
	this.getBeenHere = function()  {
		return new  FoursquareBeenHere(base["beenHere"]);
	}

	/*
	*/
	this.getCanonicalUrl = function()  {
		return base["canonicalUrl"];
	}

	/*
	Categories that have been applied to this venue
	*/
	this.getCategories = function()  {
		var listEntries = base["categories"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareCategory(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	An object containing twitter, phone, and formattedPhone of the user or venue contact. All can be present, and all can be absent from the response.
	*/
	this.getContact = function()  {
		return new  FoursquareContact(base["contact"]);
	}

	/*
	The date the venue was created in epoch format
	*/
	this.getCreatedAt = function()  {
		return base["createdAt"];
	}

	/*
	Whether or not a menu is available for this venue
	*/
	this.getHasMenu = function()  {
		return base["hasMenu"];
	}

	/*
	Contains information about who is here now
	*/
	this.getHereNow = function()  {
		return new  FoursquareHereNow(base["hereNow"]);
	}

	/*
	Get the hours that the venue is open
	*/
	this.getHours = function()  {
		return new  FoursquareHours(base["hours"]);
	}

	/*
	The id of the venue
	*/
	this.getId = function()  {
		return base["id"];
	}

	/*
	The count of users who have liked this object, and groups containing any friends and others who have liked it
	*/
	this.getLikes = function()  {
		return new  FoursquareLikes(base["likes"]);
	}

	/*
	A grouped response of lists that contain this venue
	*/
	this.getListed = function()  {
		return new  FoursquareListed(base["listed"]);
	}

	/*
	An object containing the location information for the venue
	*/
	this.getLocation = function()  {
		return new  FoursquareLocation(base["location"]);
	}

	/*
	An object containing information about the mayor of the venue
	*/
	this.getMayor = function()  {
		return new  FoursquareMayor(base["mayor"]);
	}

	/*
	An object containing url and mobileUrl that display the menu information for this venue
	*/
	this.getMenu = function()  {
		return new  FoursquareMenu(base["menu"]);
	}

	/*
	The name of the venue
	*/
	this.getName = function()  {
		return base["name"];
	}

	/*
	Get the count of page updates
	*/
	this.getPageUpdates = function()  {
		return new  FoursquarePageUpdates(base["pageUpdates"]);
	}

	/*
	Contain information about photos for this venue
	*/
	this.getPhotoGroups = function()  {
		var listEntries = getSubArrayByKey(base, "photos")["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquarePhotoGroup(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Get price information for this venue
	*/
	this.getPrice = function()  {
		return new  FoursquarePrice(base["price"]);
	}

	/*
	The venue rating
	*/
	this.getRating = function()  {
		return base["rating"];
	}

	/*
	The short url for the venue
	*/
	this.getShortUrl = function()  {
		return base["shortUrl"];
	}

	/*
	Contains information about specials offered by the venue
	*/
	this.getSpecials = function()  {
		var listEntries = base["specials"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareSpecial(listEntries[i]);
			}
		}
		return resultArray;
	}

	/*
	Contains statistics for the venue including checkinsCount, tipCount, and userCount
	*/
	this.getStats = function()  {
		return new  FoursquareStats(base["stats"]);
	}

	/*
	Contains tags for this venue
	*/
	this.getTags = function()  {
		var listEntries = base["tags"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = listEntries[i];
			}
		}
		return resultArray;
	}

	/*
	The timezone for this venue
	*/
	this.getTimeZone = function()  {
		return base["timeZone"];
	}

	/*
	Get tips from others and friends
	*/
	this.getTipGroups = function()  {
		var listEntries = getSubArrayByKey(base, "tips")["groups"];
		var resultArray = new Array();
		if(listEntries != null) {
			for(var i=0; i < listEntries.length; i++) {
				resultArray[i] = new  FoursquareTipGroup(listEntries[i]);
			}
		}
		return resultArray;
	}

	this.getTodos = function()  {
		return new  FoursquareTodos(base["todos"]);
	}

	/*
	The url for this venue
	*/
	this.getUrl = function()  {
		return base["url"];
	}

	/*
	Whether or not the venue has been verified
	*/
	this.getVerified = function()  {
		return base["verified"];
	}
}


/*
 Contains an object with a text field that contains a warning message
     
*/

var FoursquareWarning = function(base) {
	var base = base;
    if(typeof(base) == 'undefined')
        base = new Array();

	this.getBase = function() {
		return base;
	}
	/*
	The warning message
	*/
	this.getText = function()  {
		return base["text"];
	}
}
